cl_dll/ff/ff_hud_deathnotice.cpp

```diff
@@ -706,14 +706,12 @@ void CHudDeathNotice::FireGameEvent( IGameEvent * event )
 		deathMsg.iSuicide = ( !killer || ( ( killer == victim ) && ( !bBuildableKilled ) ) );
 
 		deathMsg.Assister.iEntIndex = -1;
-		int killassisterIdx = event->GetInt( "killassister", -1 );
-		if (killassisterIdx != -1)
+		int killAssister = event->GetInt( "killassister", -1 );
+		if (killAssister != -1)
 		{
-			//C_BaseEntity* pEnt = cl_entitylist->GetEnt( killassisterIdx );
-			//C_FFPlayer *pFFPlayer = ToFFPlayer( pEnt );
-			//if ( !pFFPlayer )
-			Q_strcpy( deathMsg.Assister.szName, g_PR->GetPlayerName( killassisterIdx ) );
-			deathMsg.Assister.iEntIndex = killassisterIdx;
+			killAssister = engine->GetPlayerForUserID(killAssister);

```

Comment:
I think this would be more explicit if we keep the concepts of userId and playerIndex separate and stored in different variables. It would help prevent future coders getting confused and making a bug.

killAssisterUserId and killAssisterIndex for example.